.github/composite-actions/setup-php/action.yml

```diff
@@ -0,0 +1,25 @@
+name: "Setup PHP"
+
+inputs:
+  version:
+    description: 'PHP version'
+    required: true
+  coverage:
+    description: 'coverage mode'
+    required: false
+    default: 'none'
+  tools:
+    description: 'tools'
+    required: false
+    default: 'none'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ inputs.version }}
+        coverage: ${{ inputs.coverage }}
+        tools: composer, ${{ inputs.tools }}
+        extensions: none, curl, dom, json, mbstring, opcache, openssl, simplexml, tokenizer, xml, xmlwriter, zip
```

Comment:
Do rozważenia wsparcie dla `${{ inputs.extensions }}`, ale można zastosować YAGNI i dodać to później, gdy faktycznie joby się rozjadą w kontekście wymaganych rozszerzeń.

Reply:
so far, it was the same list everywhere.
we could decide to bring parametrization when needed (or we add it directly here without parametrization, as I didn't notice it affects any timing or performance - in opposition to tools/coverage)